akka-cluster-sharding/src/main/scala/akka/cluster/sharding/Shard.scala

```diff
@@ -425,6 +465,9 @@ private[akka] trait RememberingShard { selfType: Shard ⇒
     val id = idByRef(ref)
     idByRef -= ref
     refById -= id
+    if (passivateIdleTask.isDefined) {
```

Comment:
are refById and lastMessageTimestamp always updated together? should we keep them in same `Map[String, (ActorRef, Long)]`?

Reply:
That's not a bad idea. I didn't do it like that because I wanted to keep the disabled mode to add as little overhead as possible, but maybe that would be cleaner.